components/SignUp.js

```diff
@@ -1,81 +1,160 @@
     }
-    if (
-      passwordInput.value.match(
-        /^(?=.*[0-9])(?=.*[!@#$%^&*])[a-zA-Z0-9!@#$%^&*]{6,20}$/
-      )
-    ) {
-      StrengthIndicator.textContent = "Password strength: Very strong!";
+    if (!values.verifyPassword) {
+      errors.verifyPassword = "Du måste ange samma lösenord igen!";
+    } else if (values.verifyPassword !== values.password) {
+      errors.verifyPassword = "Lösenordet överensstämmer inte!";
     }
+    return errors;
   };
 
-  // Render the signUp component.
+  // Check password strength.
+  function strengthCheck() {
+    if (formValues.password < 4) {
+      setpasswordStrength("Lösenordets styrka: Inte tillräcklig");
+    }
+    if (formValues.password.match(/^(?=.*[a-z]).{4,20}$/)) {
+      setpasswordStrength("Lösenordets styrka: Svagt");
+    }
+    if (formValues.password.match(/^(?=.*[a-z])(?=.*[A-Z]).{4,20}$/)) {
+      setpasswordStrength("Lösenordets styrka: Medelt");
+    }
+    if (formValues.password.match(/^(?=.*\d)(?=.*[a-z])(?=.*[A-Z]).{4,20}$/)) {
+      setpasswordStrength("Lösenordets styrka: Starkt");
+    }
+    if (formValues.password.match(/^(?=.*[0-9])(?=.*[!@#$%^&*])[a-zA-Z0-9!@#$%^&*]{4,20}$/)) {
+      setpasswordStrength("Lösenordets styrka: Mycket starkt");
+    }
+  }
+
+  // Post formValues to the API /api/users/newuser.
+  async function postData(formValues) {
+    const response = await fetch("http://localhost:3000/api/users/newuser", {
+      method: "POST",
+      headers: { "Content-Type": "application/json", },
+      body: JSON.stringify(formValues),
+    });
+    const data = await response.json();
+    if (data.message === "User created!") {
+      return window.location.href = "/login";
+    } else {
+      return alert(data.message);
+    }
+  }
+
+  // Render the SignUp Registration form.
   return (
-    <div className={styles["main-container-signinup"]}>
-      <div className={styles["main-container-signinup-content"]}>
-        <input id="name" type="text" placeholder="Your full name" />
-        <input id="phone" type="text" placeholder="Phone no." />
-        <input id="email" type="email" placeholder="Email address" />
-        <input onKeyUp={passwordStrength} id="password" type="password" placeholder="Password" />
-        <input id="verify-password" type="password" placeholder="Verify password" />
-        <p id="strength-indicator">Make sure to use a strong password.</p>
-        <button onClick={signUp}>Sign up</button>
-      </div>
+    <div className={styles["signup"]}>
+      <form onSubmit={handleSubmit}>
+        <div className="field">
+          <label>Ditt namn</label>
+          <input
+            type="text"
+            name="name"
+            placeholder="Förnamn + Efternamn"
+            value={formValues.name}
+            onChange={handleChange}
+          />
+        </div>
+        <p>{formErrors.name}</p>
+        <div className="field">
+          <label>Telefonnummer</label>
+          <input
+            type="number"
```

Comment:
Phone numbers are not "numbers" in this sense of the word. Inputs for phone numbers should use `text`, so that users can type leading zeroes etc, or some sort of special widget to handle phone number formatting.